base/strings/util.jl

```diff
@@ -77,9 +79,31 @@ julia> a = "March"
 
 julia> chop(a)
 "Marc"
+
+julia> chop(a, 1, 2)
+"ar"
+
+julia> chop(a, 5, 5)
+""
 ```
 """
-chop(s::AbstractString) = SubString(s, 1, prevind(s, endof(s)))
+function chop(s::AbstractString, head::Integer, tail::Integer)
+    # negative values of head/tail will throw error in nextind/prevind
+    if head == 0
+        if tail == 0
+            return SubString(s)
```

Comment:
Wouldn't it be simpler to compute the start and end indices in the branches, and call `SubString` only once at the end? `SubString` will call `endof` anyway if you don't do it yourself. That should make the function body smaller and allow inlining when appropriate.

Reply:
an excellent suggestion. Thank you.

Reply:
Wow, indeed that's compact!